evaluate_iteration_changes.py

```diff
@@ -0,0 +1,131 @@
+)
+import json
+import pprint
+import editdistance
+from src.metrics import rouge
+
+pp = pprint.PrettyPrinter(indent=2)
+
+
+def collect_iteration_stats(
+    logs_path,
+    xsum_test,
+    test_set_ids: Union[set, None],
+    entity_match_type: EntityMatchType = "contained",
+    should_annotate=False,
+):
+    is_oracle = "oracle" in logs_path
+    gold_sums, gold_metadata = get_gold_xsum_data()
+
+    with open(logs_path, "r") as f:
+        logs = json.load(f)
+
+    sorted_keys = sorted([int(x) for x in logs["iterations"].keys()])
+
+    iteration_stats = []
+
+    sums_by_id = {}
+    sum_ents_by_id = {}
+    baseline_eval_sums_by_id = {}
+    for iteration_idx in sorted_keys:
+        iteration_data = logs["iterations"][str(iteration_idx)]
+        summaries = iteration_data["summaries"]
+        current_iteration_stats = {
+            "iteration": iteration_idx,
+            "summary": {},
+            "summary_remaining": 0,
+        }
+        updated_sums_by_id = {}
+        for sum_id, data in summaries.items():
+            if sum_id in test_set_ids or test_set_ids is None:
+                updated_sums_by_id[sum_id] = data["summary"]
+                sums_by_id[sum_id] = data["summary"]
+                sum_ents_by_id[sum_id] = data["labeled_entities"]
+                current_iteration_stats["summary_remaining"] += 1
+                current_iteration_stats["summary"][sum_id] = {
+                    "summary": data["summary"]
+                }
+        if current_iteration_stats["summary_remaining"] == 0:
+            break
+        stats_factuality, eval_sums_by_id = evaluate_factuality(
+            sums_by_id,
+            sum_ents_by_id,
+            gold_sums,
+            gold_metadata,
+            xsum_test,
+            should_annotate,
+            entity_match_type=entity_match_type,
+            print_first_n=0,
+            is_fbs=True,
+            is_oracle=is_oracle,
+            compute_rouge=False,
+        )
+        current_iteration_stats["factuality_summary"] = stats_factuality["summaries"]
+        current_iteration_stats["factuality_entities"] = stats_factuality["entities"]
+        current_iteration_stats["summary_completed"] = (
+            stats_factuality["summaries"]["total"]
+            - current_iteration_stats["summary_remaining"]
+        )
+
+        # compute edit stats
+        if iteration_idx == 0:
+            baseline_eval_sums_by_id = eval_sums_by_id
+        if iteration_idx > 0:
+            for sum_id, sum in updated_sums_by_id.items():
+                if sum != SUMMARY_FAILED_GENERATION:
+                    original_sum = baseline_eval_sums_by_id[sum_id]["summary"]
+                    rouge_scores = rouge([sum], [original_sum])
+                    current_iteration_stats["summary"][sum_id]["edit_stats"] = {
+                        "edit_distance_token": editdistance.eval(
+                            original_sum.lower().split(" "), sum.lower().split(" ")
```

Comment:
so because we're splitting on whitespace, we're doing word-level edit distance? I think you mentioned on our call that it was token-level

Reply:
Yeah, it's word-level